### ISSUE TITLE
{N} 2.4.0 & webpack: css files missing

### ISSUE THREAD

**Issue by fleuverouge** [NativeScript/nativescript-dev-webpack#27]
Everything worked in 2.3.0

My component:

```
@Component({
    selector: 'DatePickerView',
    styleUrls: ["UIElements/date-picker-view.css"],
    template: require('./date-picker-view.html'),
})

export class DatePickerView implements AfterViewInit {...}
```

webpack.config.js:

```
var bundler = require("nativescript-dev-webpack");
var path = require("path");

module.exports = bundler.getConfig({
    resolveLoader: {
        root: path.join(__dirname, "node_modules")
    },
    module: {
        loaders: [
            {
                test: /\.html$/,
                loader: "html"
            },
        ]
    }
});
```

What I run:

```
tns platform add ios
tns prepare ios --bundle
tns emulate ios --bundle
```

I got:

> CONSOLE ERROR file:///app/bundle.js:8939:31: ns-renderer: ERROR BOOTSTRAPPING ANGULAR
> CONSOLE ERROR file:///app/bundle.js:8939:31: ns-renderer: File \<App location\>/Test.app/app/UIElements/date-picker-view.css does not exist. Resolved from: UIElements/date-picker-view.css.
> 	
> 	get@file:///app/bundle.js:79158:29
> 	_fetch@file:///app/bundle.js:47474:49
> 	file:///app/bundle.js:47574:62
> 	map@[native code]
> 	_loadMissingExternalStylesheets@file:///app/bundle.js:47574:20
> 	normalizeExternalStylesheets@file:///app/bundle.js:47556:55
> 	invoke@file:///app/bundle.js:32155:34
> 	run@file:///app/bundle.js:32048:50
> 	file:///app/bundle.js:32414:61
> 	invokeTask@file:///app/bundle.js:32188:43
> 	runTask@file:///app/bundle.js:32088:58
> 	drainMicroTaskQueue@file:///app/bundle.js:32320:43
> 	promiseReactionJob@[native code]
> 	UIApplicationMain@[native code]
> 	start@file:///app/bundle.js:6083:31
> 	bootstrapModule@file:///app/bundle.js:199:27
> 	file:///app/bundle.js:93:58
> 	__webpack_require__@file:///app/bundle.js:21:34
> 	file:///app/bundle.js:41:37
> 	anonymous@file:///app/bundle.js:42:12
> 	evaluate@[native code]
> 	moduleEvaluation@[native code]
> 	[native code]
> 	promiseReactionJob@[native code]

_<Project location>/platforms/ios/build/emulator/Test/app_ only includes: App_Resources, bundle.js & package.json

The same problem goes to html files if I don't use html loader. I tried css loader but another error occurred:

> file:///app/bundle.js:86527:37: JS ERROR ReferenceError: Can't find variable: window

Any suggestion is appreciated <3

**Comment by corne-de-bruin** [comment]
Since we've upgraded to {N} 2.4 (from 2.2) and to Angular 2.2.1 we are facing the same issue with webpack. 

So far we haven't found a solution yet for importing the css files.

**Comment by fleuverouge** [comment]
After having tried some loaders, I found raw loader work. However app.css seemed to be ignored :( And if I imported theme file, eg:

`@import 'nativescript-theme-core/css/core.light.css';
`

As expected, that "file does not exist" error was thrown:

> CONSOLE ERROR file:///app/bundle.js:8880:31: ns-renderer: File \<App location\>/TestWebpack.app/app/nativescript-theme-core/css/core.light.css does not exist. Resolved from: nativescript-theme-core/css/core.light.css.

**Comment by fleuverouge** [comment]
Update:
I managed to make the styles work (without @import though): Add app.css to the styles of the main component (eg AppComponent), and set its encapsulation to ViewEncapsulation.None in order to allow styles from outside components to be applied. Samples:

```
import { Component, ViewEncapsulation } from "@angular/core";

@Component({
    selector: "my-app",
    template: `<page-router-outlet></page-router-outlet>`
    encapsulation: ViewEncapsulation.None,
    styles: [require("./app.css"), require("./app.component.css")],
})
export class AppComponent { ... }
```

Please note that this will make any child component reuse its parent's style.

**Comment by hdeshev** [comment]
Styles are a pretty complex to get right. I've been working on a new approach to bundling that will be somewhat backwards incompatible, but at least I got css bundling (including theme imports) working. I noticed we have two types of CSS files:

1. app.css -- that one is best processed by the webpack ExtractTextPlugin so we can generate a CSS bundle at the app root.
2. component-specific CSS files -- those are easiest to embed using the raw-loader (and perhaps the angular2-template-loader, so that you don't have to switch `templateUrl` settings to `template` ones manually.

Perhaps [this commit](https://github.com/NativeScript/nativescript-dev-webpack/commit/2a71d03f25827794d2b718964765c7e24e2bad0b#diff-c7ff561b55a2e7743b7e98fdf1af84deR59) could help, if you need to support that before us releasing the new version of the plugin.

**Comment by fleuverouge** [comment]
Hi, @hdeshev ,
Thanks for your reply.
I had to use css loader for component-specific css files because some of them import other files. So I use css-to-string-loader to add them into the common bundle.
When using ExtractTextPlugin with app.css to generate css bundle, somehow in the _/platforms/ios/build/emulator/Test/app_ folder I got app.css and a weird app.css.js file:

```
module.exports =
/******/ (function(modules) { // webpackBootstrap
/******/ 	// The module cache
/******/ 	var installedModules = {};

/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {

/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId])
/******/ 			return installedModules[moduleId].exports;

/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			exports: {},
/******/ 			id: moduleId,
/******/ 			loaded: false
/******/ 		};

/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);

/******/ 		// Flag the module as loaded
/******/ 		module.loaded = true;

/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}


/******/ 	// expose the modules object (__webpack_modules__)
/******/ 	__webpack_require__.m = modules;

/******/ 	// expose the module cache
/******/ 	__webpack_require__.c = installedModules;

/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = "";

/******/ 	// Load entry module and return exports
/******/ 	return __webpack_require__(0);
/******/ })
/************************************************************************/
/******/ ([
/* 0 */
/*!*****************!*\
  !*** ./app.css ***!
  \*****************/
/***/ function(module, exports) {

	// removed by extract-text-webpack-plugin

/***/ }
/******/ ]);
```

**Comment by hdeshev** [comment]
I get the weird app.css.js file too, but since it's not used at runtime I decided to keep it around since the other option would be to ask developers to add a `require` call for their app.css file somewhere in their code.

We published nativescript-dev-webpack@0.2.11 to npm today with all the changes. You could try that, but remember to delete all your webpack configs since the initial `npm install` will create the default config from a predefined template.

**Comment by hdeshev** [comment]
Closing this old issue. Will reopen, if this is still a problem.
